### PR TITLE
use const references where appropriate

### PR DESCRIPTION
**Pull request by Diapolo**

### CODE REVIEW

**Comment by laanwj** [comment]
There are still a *lot* of these cases where a const reference can be used for strings throughout the source code. Instead of many little pulls I'd prefer (for post-0.11) a pull where they're all addressed at once.

A quick `git grep string -- \*.h|grep -v '&'` shows at least:
- FormatParagraph
- CAlert::AppliesTo
- ParseScript()
- GetWarnings()
- ParseNetwork()
- CRPCTable::operator[](std::string name)
- CRPCTable::help(std::string name)
- HelpExampleCli
- HelpExampleRpc
- runCommand()
- CWallet::GetAccountAddresses
- CWallet::CWallet(std::string strWalletFileIn)
- CValidationState::Error
- `CNode(SOCKET hSocketIn, CAddress addrIn, std::string addrNameIn = "", bool fInboundIn=false)`

**Comment by Diapolo** [comment]
I'm going to update this :).

**Comment by Diapolo** [comment]
ParseNetwork() isn't suitable, because we use ``boost::to_lower(net);`` inside.

**Comment by Diapolo** [comment]
@laanwj I catched all your results and also changed a few other occurances in BOOST_FOREACH loops and in the CNode constructor (unrelated to std::string). Mind reviewing it?

File: src/net.cpp [hunk 1]
```diff
@@ -1906,7 +1906,7 @@ bool CAddrDB::Read(CAddrMan& addr)
 unsigned int ReceiveFloodSize() { return 1000*GetArg("-maxreceivebuffer", 5*1000); }
 unsigned int SendBufferSize() { return 1000*GetArg("-maxsendbuffer", 1*1000); }
 
-CNode::CNode(SOCKET hSocketIn, CAddress addrIn, std::string addrNameIn, bool fInboundIn) :
+CNode::CNode(const SOCKET& hSocketIn, const CAddress& addrIn, const std::string& addrNameIn, bool fInboundIn) :
```
Comment (laanwj) [inline src/net.cpp]: No need to do this for SOCKET, which is just an integer handle.

Comment (Diapolo) [inline src/net.cpp]: You are right, reverted for hSocketIn :), thanks.

**Comment by Diapolo** [comment]
@laanwj I'm going to attach a second squashme pull, which covers BOOST_FOREACH and updated to constant refs if appropriate, if you want me to do so?

**Comment by laanwj** [comment]
utACK

**Comment by laanwj** [comment]
> I'm going to attach a second squashme pull, which covers BOOST_FOREACH and updated to constant refs if appropriate, if you want me to do so?

Yes, why not

**Comment by Diapolo** [comment]
@laanwj See 2nd commit.

**Comment by laanwj** [comment]
utACK^2 . WIll merge this after #6121

**Comment by Diapolo** [comment]
That's fine, could be I forgot a few cases in the GUI code, but that can be a seperate pull.

**Comment by fanquake** [comment]
utACK

**Comment by laanwj** [comment]
Will merge after rebase.
@diapolo can you please include `getValStr()` https://github.com/bitcoin/bitcoin/blob/master/src/univalue/univalue.h#L64 to return a const reference?

**Comment by Diapolo** [comment]
@laanwj Fixed 2 merge conflicts, one in rpcnet.cpp and the other in utilstrencodings.cpp and added getValStr(). Should be ready.

**Pull request merged**